<!--
%\VignetteEngine{knitr::knitr}
%\VignetteIndexEntry{Introduction to antiProfiles}
-->

Introduction to Gene Expression Anti-profiles
========================================================

This package implements the gene expression anti-profiles method in [Corrada Bravo, et al.][ap]. In this vignette we will use the companion `antiProfilesData` to illustrate some of the analysis in that paper. 

Colon cancer expression data
-----------------------------

The `antiProfilesData` package contains expression data from normal colon tissue samples and colon cancer samples. It contains data from two datasets in the Gene Expression Omnibus, [Marra et al.](http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE8671) and [Gyorffy, et al.](http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE4183). Probesets annotated to genes within blocks of hypo-methylation in colon cancer defined in [Hansen, et al.][kh]. 

Let's load the data and take a look at its contents.

```{r}
require(antiProfilesData)
data(apColonData)
show(apColonData)

# look at sample types by experiment and status
table(apColonData$Status, apColonData$SubType, apColonData$ExperimentID)
```

The data is stored as an `ExpressionSet`. This dataset contains colon adenomas, benign but hyperplastic growths along with the normal and tumor tissues. Let's remove these from the remaining analysis.

```{r}
drop=apColonData$SubType=="adenoma"
apColonData=apColonData[,!drop]
```

Building antiprofiles
---------------------------------------

We will use data from one of the experiments to train the anti-profile and test it on the data from the other experiment.

### Computing variance ratios

The first step in building an antiprofile is to calculate the ratio of normal variance to cancer variance. This is done with the `apStats` function.

```{r}
require(antiProfiles)
trainSamples=pData(apColonData)$ExperimentID=="GSE4183"
colonStats=apStats(exprs(apColonData)[,trainSamples], pData(apColonData)$Status[trainSamples],minL=5)
show(colonStats)
```

We can see how that ratio is distributed for these probesets:

```{r, fig.width=4, fig.height=4}
hist(colonStats@probes$stat, nc=100, main="Histogram of log variance ratio", xlab="log2 variance ratio")
```

### Building the anti-profile

Now we construct the anti-profile by selecting the 100 probesets most hyper-variable probesets

```{r}
ap=buildAntiProfile(colonStats, tissueSpec=FALSE, sigsize=100)
show(ap)
```

### Computing the anti-profile score

Given the estimated anti-profile, we can get anti-profile scores for a set of samples.

```{r, fig.width=6, fig.height=4}
counts=apCount(ap, exprs(apColonData)[,!trainSamples])
require(RColorBrewer)
palette(brewer.pal(8,"Dark2"))
o=order(counts)
dotchart(counts[o],col=pData(apColonData)$Status[!trainSamples][o]+1,labels="",pch=19,xlab="anti-profile score", ylab="samples",cex=1.3)
legend("bottomright", legend=c("Cancer","Normal"),pch=19,col=2:1)
```

We see in this case that the anti-profile score can distinguish the test samples perfectly.

References
-----------

- Corrada Bravo, H., Pihur, V., McCall, M., Irizarry, R.A., Leek, J.T. (2012). "Gene expression anti-profiles as a basis for accurate universal cancer signatures" BMC Bioinformatics, 13:272 [BMC Bioinformatics][ap]  

- Hansen, K. D., Timp, W., Bravo, H. C., Sabunciyan, S., Langmead, B., McDonald, O. G., Wen, B., et al. (2011). "Increased methylation variation in epigenetic domains across cancer types." Nature Genetics, 43(8), 768â€“775. [Nat. Genetics][kh]


[ap]: http://www.biomedcentral.com/1471-2105/13/272 "Gene Expression Anti-Profiles"
[kh]: http://www.nature.com/ng/journal/v43/n8/full/ng.865.html "Domains of methylation loss in colon cancer"




